src/lib.rs

```diff
@@ -16,11 +25,77 @@
 use std::convert::From;
 use std::io::{Result, Write};
 use std::mem;
+use std::ops;
+use std::fmt;
```

Comment:
Import here the traits and structs that you use down below and sort the `use` statements alphabetically.